### PR TITLE
[13.0][MIG] product_pack

### PR DESCRIPTION
**Pull request by jjscarafia**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please see Travis status.

**Comment by jjscarafia** [comment]
Hi @pedrobaeza thanks for the review. Can you help me a little bit? It seams to bre related to new pre-commit implementation but I didn't found documentation regarding this.
1. I've already run the "pre-commit run -a" as pointed [here](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch)
2.Some lint errors I can fix and push again, but I don't understand [this one](https://travis-ci.com/github/OCA/product-pack/jobs/317894083#L321)


Thanks!

**Comment by jjscarafia** [comment]
fixed @pedrobaeza 
thanks!

**Review by pedrobaeza — APPROVED** [review]

File: product_pack/models/product_pack_line.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @api.constrains("product_id")
+    def _check_recursion(self):
+        """Check recursion on packs."""
+        for line in self:
+            parent_product = line.parent_product_id
+            pack_lines = line
+            while pack_lines:
+                if parent_product in pack_lines.mapped("product_id"):
+                    raise ValidationError(
+                        _("You cannot set recursive packs.\n" "Product id: %s")
```
Comment (ernestotejeda) [inline product_pack/models/product_pack_line.py]:
```suggestion
                        _("You cannot set recursive packs.\nProduct id: %s")
```

File: product_pack/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 NaN (http://www.nan-tic.com) - Àngel Àlvarez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Product Pack",
+    "version": "13.0.1.0.0",
+    "category": "Product",
+    "summary": "This module allows you to set a product as a Pack",
+    "website": "https://github.com/OCA/product-pack",
+    "author": "NaN·tic, " "ADHOC SA, " "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (ernestotejeda) [inline product_pack/__manifest__.py]:
```suggestion
    "author": "NaN·tic, ADHOC SA, Tecnativa, Odoo Community Association (OCA)",
```
I know that black does things like this but they look a bit ugly and if we fixes it like that black won't change it again

**Review by ernestotejeda — CHANGES_REQUESTED** [review]
Thanks @jjscarafia :+1: 

- Remove product_pack/migrations/12.0.1.0.0 folder

**Comment by pedrobaeza** [comment]
Please include #15

**Comment by pedrobaeza** [comment]
Please include as well #18

File: product_pack/models/product_pack_line.py (hunk 1 above)
Comment (jjscarafia) [reply product_pack/models/product_pack_line.py]: done @ernestotejeda thanks!

**Comment by jjscarafia** [comment]
> Thanks @jjscarafia 
> 
> * Remove product_pack/migrations/12.0.1.0.0 folder

I're removed them but I don't get the point of doing it. We use odoo upgrade service and we often jump versions, for eg. we send odoo a v11 db and we get a v13 one, and we would like migration scriptf for v12 to be run while upgrading to v13

Any opinion regarding this @ernestotejeda @pedrobaeza ?

**Comment by jjscarafia** [comment]
all the changes done. Thanks @pedrobaeza @ernestotejeda 

@pedrobaeza https://github.com/OCA/product-pack/pull/18/files will be considered on new pr for stock_product_pack

**Comment by jjscarafia** [comment]
I've closed this PR and open a new one with same content because travis was red and I couldn't achive to re run it without a new PR.
https://github.com/OCA/product-pack/pull/22

**Pull request closed without merging**